### ISSUE TITLE
SpamAssassin configuration files per mailbox and domain

### ISSUE THREAD

**Issue by ADDISON74** [virtualmin/virtualmin-gpl#617]
After installing Virtualmin and creating a virtual server in the **/home/[domain]** directory, there is no **.spamassassin** directory. This is a correct behavior.

If we go to the UI ```Virtualmin > Email Settings > Spam and Virus Scanning``` and choose Yes for the **Allow mailbox users to create mail filters option**, then the **/.spamassassin** directories are created for the domain itself and for each individual email address. This directory contains three files, including **user_prefs** which is read last by SpamAssassin.

The order of reading the configuration files is from top to bottom in the following directories

**/usr/share/spamassassin**

**/etc/mail/spamassassin**
(note: in Debian this directory is a symlink to /etc/spamassassin)

**/etc/webmin/virtual-server/spam/[vserver_id]**

(note: in this directory there are symlinks to the directory above. there is a file named virtualmin.cf which is read last)

**/home/[domain]/homes/[email_box]/.spamassassin**

If in the meantime the previously mentioned option is set to No, the **/.spamassassin** directories remain created in /home directory, which is not right because SpamAssassin will still continue read the **user_prefs** file and apply the configuration. I noticed this when I changed the score in the first three directories above, but the one in /home was always taken over.

I consider this a bug. When the option is set to No, all /.spamassassin directories in /home/[domain] must be deleted (if exists) to prevent overwriting the configuration, and thus the last file read will be **virtualmin.cf**.

![spamassassin](https://github.com/virtualmin/virtualmin-gpl/assets/8360474/6eb836a6-29ae-4f71-8845-a3083a184e2f)

**Comment by ADDISON74** [comment]
These tests were done in Debian 10. Now I discovered that in Debian 12 the .spamassassin directories are not created when the option is changed to Yes (still investigating before reporting as a new issue).

**Comment by jcameron** [comment]
So the "Allow mailbox users to create mail filters option" option controls whether users can have procmail rules - it's actually unrelated to per-user spamassassin configs.

**Comment by ADDISON74** [comment]
@jcameron - I agree with you that it is very clear from the option tooltip what its purpose is. But I have identified an issue, namely that the use of this option has effects that I did not know about yet. When it is set to Yes, SpamAssassin creates in home and user mailboxes its own directories which, if used and then the option is set to No, it will influence the functionality. Actually the configuration from the user_prefs files is still being applied.
